sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -220,7 +220,21 @@ case class GpuElementAt(left: Expression, right: Expression, failOnError: Boolea
           }
         }
       case _: MapType =>
-        (_, _) => throw new UnsupportedOperationException("non-literal key is not supported")
+        (map, indices) => {
+          if (failOnError) {
+            withResource(map.getMapKeyExistence(indices)) { keyExists =>
+              withResource(keyExists.all()) { exist =>
+                if (!exist.isValid && exist.getBoolean) {
+                  map.getMapValue(indices)
+                } else {
+                  throw new NoSuchElementException("One of the keys doesn't exist")
```

Comment:
nitL it would be nice to use the shim layer for this kind of error like what spark is doing just in case they decide to change it on us in the future. Also can we pick the first key that was missing? we already know we are going to throw an error, so performance is not critical.

```
  def mapKeyNotExistError(key: Any): NoSuchElementException = {
    new NoSuchElementException(s"Key $key does not exist.")
  }
```